### PR TITLE
fix, refactor and optimization value package

### LINKED ISSUE #1542: permanently stuck on zoomed out if navigation console gets blown up
## Round ID: 2779

unknown 1

### LINKED ISSUE #1624: how to get on the overmap
## Testmerges:

#1586: 'Makes superstructure crit slightly more explosive' by Bokkiewokkie at commit 0a32ff201a
#1600: 'Inertial Dampener tgui' by TheOneAndMightyRed at commit 573b67448c
#1574: 'Adds revenant z-level moving' by Golubian at commit 93e0254b68
#1614: 'Gamemode round-end tweaks' by covertcorvid at commit ad74b760e2
#1612: 'Fixes creation of phantom stormdrive control rods that don't degrade' by IndusRobot at commit 4e5a051304
#1553: 'Laser Anti-missile System for solgov + one refactor for automated weapons ' by IndusRobot at commit 991d741846
#1622: 'KNPC Tweaks' by Karmic-Skink at commit ca9629a865
#1618: '"Fixes" some homing stuff, plus more nukings' by DeltaFire15 at commit c5a582c426
#1615: 'Fixes indestructible extra loading rack when gauss gun is destroyed and rebuilt' by covertcorvid at commit 2bbfe66395
#1607: 'world initialization runtime fixes + lazylift and other misc optimizations' by TheFakeElon at commit 8bf3a51dcc
#1620: 'Makes the title screen do widescreen good (assuming the file is the right size)' by Bokkiewokkie at commit 3376127043
#1617: 'No-kill traitor option' by covertcorvid at commit e021e459c3

## Reproduction:

step 1: get in fighter

step 2: fly fighter onto the overmap

step 3: make sure your cockpit canopy is OPEN

step 4: take your ID out and swipe your ID on your own fighter

step 5: click the funny eject button

you are now on the overmap

### LINKED ISSUE #1631: Another going to overmap bug.
As a pilot the Spatial instability mutation will throw you out of a fighter into the overmap if your fighter is there as it does it's teleport. The mutation can for example be acquired from a radstorm.

### LINKED ISSUE #1632: Yet another lets go on the overmap bug
## Round ID:
3033

## Testmerges:

#1586
#1574 
#1614 
#1612 
#1553 
#1622 
#1618 
#1615 
#1607 
#1620 
#1617 
#1628 

## Reproduction:

Step 1: get bluespace body bag
Step 2: have pilot take it to overmap
Step 3: deploy bodybag
Step 4:
grafik
Chainsaw goes brrrrr

### PR DESCRIPTION
**Pull request by TheFakeElon**
fixes #1624
fixes #1631
fixes #1542
fixes #1632

## About The Pull Request

tldr; optimizes a bunch of shitcode and fixes a few issues

players can no longer enter the overmap by ejecting in maintenance mode with their canopy open (#1624)

players can no longer enter the overmap with bluespace bags (#1632)

Player screens should no longer be stuck in unusual perspectives when their ship monitoring console is destroyed (#1542)

Optimizes and fixes a significant portion of of fighters.dm

ship weapons now have added intermediate weapon states to compensate for sleep() calls

Fixes a lot of improper capital names

Shields no longer spam the server with stop lag calls in a while(exists) loop **every** time they're hit with a projectile

AI ship Cloaking and decloaking uses much more efficient animation

## Changelog
:cl:
fix: Fixes multiple bugs that allowed mobs to enter the overmap
fix: Fixes client screen bug when overmap consoles are destroyed
code: Somewhat significant overmap and overlay optimizations
code: General name fixes and code cleaning
code: Optimizes projectile movement slightly
refactor: refactors fighter overlay and turf move handling
refactor: refactors ship weapon intermediate states, reloading should be smoother now
/:cl:

### CODE REVIEW

**Comment by Pontenerd** [comment]
Elon really does just come in here, fix a bunch of the spaghetti, then dip for a few months, absolute legend

**Comment by TheFakeElon** [comment]
adding fix for #1632

File: nsv13/code/modules/overmap/fighters/_fighters.dm [hunk 1]
```diff
@@ -532,54 +536,20 @@ Been a mess since 2018, we'll fix it someday (probably)
 	//check_overmap_elegibility(TRUE)
 
 /obj/structure/overmap/fighter/proc/throw_pilot() //Used when yeeting a pilot out of an exploding ship
+	var/m_damage = 200
```
Comment (Bokkiewokkie) [inline nsv13/code/modules/overmap/fighters/_fighters.dm:539]: You might want to make this part of the proc as `throw_pilot(damage = 200)` so we can modify it if we want to use it for less serious accidents. (or make it not kill the pilot)

File: nsv13/code/game/gamemodes/pvp/items.dm [hunk 2]
```diff
@@ -11,7 +11,7 @@
 	icon_state = "warden" //placeholder
 
 /obj/machinery/computer/cargo/express/syndicate
-	name = "Syndicate Requisitions Console"
+	name = "syndicate Requisitions Console"
```
[the remark this answers is not in the document]
Comment (TheFakeElon) [reply nsv13/code/game/gamemodes/pvp/items.dm:14]: oops!

File: nsv13/code/game/machinery/computer/munitions.dm [hunk 3]
```diff
@@ -68,18 +61,22 @@
...
 	switch(action)
 		if("toggle_load")
 			if(SW.state == STATE_LOADED)
 				SW.feed()
 			else
 				SW.unload()
 		if("chamber")
-			SW.chamber()
+			if(SW.state == STATE_CHAMBERING)
+				to_chat(usr, "<span class='warning'>You can't do that right now.</span>")
```
[the remark this answers is not in the document]
Comment (TheFakeElon) [reply nsv13/code/game/machinery/computer/munitions.dm:77]: Chambering covers both chambering and unchambering, but I'll make it more descriptive

File: nsv13/code/__HELPERS/overmap.dm [hunk 4]
```diff
@@ -21,3 +21,35 @@ Helper method to get what ship an observer belongs to for stuff like parallax.
...
+/// Finds a turf outside of the overmap
+/atom/proc/GetSafeTurf()
+	if(!SSmapping.level_trait(z, ZTRAIT_OVERMAP))
+		return get_turf(src)
+
+	var/max = world.maxx - TRANSITIONEDGE
+	var/min = TRANSITIONEDGE + 1
+	var/list/possible_transitions = list()
+	for(var/datum/space_level/D as() in SSmapping.z_list)
+		if(D.linkage == CROSSLINKED && !D.traits[ZTRAIT_OVERMAP])
```
Comment (covertcorvid) [inline nsv13/code/__HELPERS/overmap.dm:34]: we don't keep the empty space levels around, so we might not need the crosslinked check

File: nsv13/code/modules/mob/mob.dm [hunk 5]
```diff
@@ -4,3 +4,9 @@
 /mob/Move(atom/newloc, direct=0)
 	. = ..()
 	find_overmap()
+
+// There's so many edge cases for player overmap escapes that it's just cleaner to universally check mob forcemoves.
+// use doMove if you must get around this for whatever reason
+/mob/living/forceMove(atom/destination)
```
Comment (covertcorvid) [inline nsv13/code/modules/mob/mob.dm:10]: bless

File: nsv13/code/modules/overmap/radar.dm [hunk 6]
```diff
@@ -89,13 +89,13 @@ Called by add_sensor_profile_penalty if remove_in is used.
 	add_sensor_profile_penalty(max_tracking_range, RADAR_VISIBILITY_PENALTY)
 
 /obj/machinery/computer/ship/dradis/proc/send_radar_pulse()
-	var/obj/structure/overmap/OM = get_overmap()
 	if(!can_radar_pulse())
 		return FALSE
-	OM?.send_radar_pulse()
+	var/obj/structure/overmap/OM = get_overmap()
```
Comment (covertcorvid) [inline nsv13/code/modules/overmap/radar.dm:94]: get_overmap can actually fail though

**Review by covertcorvid — APPROVED** [review]

File: nsv13/code/modules/overmap/radar.dm (hunk 6 above)
Comment (TheFakeElon) [reply nsv13/code/modules/overmap/radar.dm:94]: Right! Right. Not sure why I did that 🤔

**Comment by covertcorvid** [comment]
Marking as high priority because overmap teleport bugs and exploits are terrible

File: nsv13/code/modules/mob/mob.dm [hunk 7]
```diff
@@ -4,3 +4,9 @@
...
+	destination = GetSafeTurf()
```
Comment (covertcorvid) [inline nsv13/code/modules/mob/mob.dm:11]: should this be destination.GetSafeTurf() ?
it looks like if someone was not on the overmap, then lines 27 and 28 would just teleport them to the turf they're currently on, right?

Comment (TheFakeElon) [reply nsv13/code/modules/mob/mob.dm:11]: *scream

I forgot I moved the order around. Cheers

**Comment by covertcorvid** [comment]
I can't point to the exact line yet but this is breaking the guns firing

File: nsv13/code/modules/overmap/fighters/_fighters.dm [hunk 8]
```diff
@@ -542,55 +550,20 @@ Been a mess since 2018, we'll fix it someday (probably)
 	forceMove(get_turf(locate(250, y, z)))
 	//check_overmap_elegibility(TRUE)
 
-/obj/structure/overmap/fighter/proc/throw_pilot() //Used when yeeting a pilot out of an exploding ship
+/obj/structure/overmap/fighter/proc/throw_pilot(m_damage = 200) //Used when yeeting a pilot out of an exploding ship
```
Comment (Bokkiewokkie) [inline nsv13/code/modules/overmap/fighters/_fighters.dm:553]: Just calling this var `damage` makes way more sense

Comment (TheFakeElon) [reply nsv13/code/modules/overmap/fighters/_fighters.dm:553]: I called it that so it wouldn't conflict with anything in the future as damage is a very simple name.
stands for mob damage

Comment (TheFakeElon) [reply nsv13/code/modules/overmap/fighters/_fighters.dm:553]: this is PEDANTIC which is supposed to be MY thing but I'll concede.. this time 🔪

File: nsv13/code/modules/munitions/ship_weapons/_ship_weapon.dm [hunk 9]
```diff
@@ -344,7 +340,9 @@
  * Transitions from STATE_LOADED to STATE_FED
  */
 /obj/machinery/ship_weapon/proc/feed()
+	set waitfor = FALSE
```
Comment (covertcorvid) [inline nsv13/code/modules/munitions/ship_weapons/_ship_weapon.dm:343]: Setting waitfor=FALSE here is causing weapons which automatically chamber to fail, because chamber gets tried before this is actually done

**Review by covertcorvid — CHANGES_REQUESTED** [review]
The shield overlay and the canopy overlays don't show up.

On the plus side, the weapons look to be working now, the forceMove change seems to work well, and overmap movement, stop_piloting, cargo missions, admin holograms, and the bloodling are all fine.

File: nsv13/code/modules/overmap/radar.dm [hunk 10]
```diff
@@ -297,8 +297,7 @@ Called by add_sensor_profile_penalty if remove_in is used.
 			var/newDelay = input(usr, "Set a new radar delay (seconds)", "Radar Delay", null) as num|null
 			if(!newDelay)
 				return
-			newDelay = newDelay SECONDS
-			newDelay = CLAMP(newDelay, MIN_RADAR_DELAY, MAX_RADAR_DELAY)
+			newDelay = CLAMP(newDelay, MIN_RADAR_DELAY, MAX_RADAR_DELAY) SECONDS
```
[the remark this answers is not in the document]
Comment (TheFakeElon) [reply nsv13/code/modules/overmap/radar.dm:300]: Oops! They weren't, I meant to put seconds inside with `newDelay`

File: nsv13/code/modules/overmap/overmap.dm [hunk 11]
```diff
@@ -456,24 +459,23 @@ Proc to spool up a new Z-level for a player ship and assign it a treadmill.
...
+	if(!SSmapping.level_trait(destination.z, ZTRAIT_OVERMAP))
 		return //No :)
 	return ..()
 
 /obj/structure/overmap/proc/find_area()
 	if(role == MAIN_OVERMAP) //We're the hero ship, link us to every ss13 area.
 		for(var/X in GLOB.teleportlocs) //Teleportlocs = ss13 areas that aren't special / centcom
-			var/area/area = GLOB.teleportlocs[X] //Pick a station area and yeet it.
-			linked_areas += area
+			var/area/A = teleportlocs[X]
```
Comment (covertcorvid) [inline nsv13/code/modules/overmap/overmap.dm:477]:
```suggestion
			var/area/A = GLOB.teleportlocs[X]
```

File: nsv13/code/modules/overmap/fighters/_fighters.dm [hunk 12]
```diff
@@ -1730,24 +1691,25 @@ Utility modules can be either one of these types, just ensure you set its slot t
...
+	if(QDELETED(C))
+		canopy.icon_state = "canopy_missing"
 		return
 	if(C.obj_integrity <= 20)
-		add_overlay(image(icon = icon, icon_state = "canopy_breach", dir = 1))
+		canopy.icon_state = "canopy_breach"
 		return
 	if(canopy_open)
-		add_overlay("canopy_open")
+		canopy.icon_state = "canopy_open"
```
Comment (covertcorvid) [inline nsv13/code/modules/overmap/fighters/_fighters.dm:1712]: I haven't used mutable_appearance enough to know why, but the canopy icons aren't updating in game. Also the canopy being closed and undamaged is how the actual fighter is drawn, so the overlay needs to be either removed or invisible for that state.

**Review by covertcorvid — CHANGES_REQUESTED** [review]

**Review by covertcorvid — APPROVED** [review]
